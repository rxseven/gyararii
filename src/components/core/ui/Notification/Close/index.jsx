import React from 'react';

import './Close.styles.scss';

function Close({ closeToast }) {
  return (
    <span
      aria-hidden="true"
      className="notification-close"
      onClick={closeToast}
    >
      &times;
    </span>
  );
}

export default Close;
