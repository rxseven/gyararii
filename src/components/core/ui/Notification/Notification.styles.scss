// Plugin styles
// https://fkhadra.github.io/react-toastify

@import "~react-toastify/dist/ReactToastify.css";

.notification-toast {
  background-color: #f8d7da;
  border: #f5c6cb solid 1px;
  border-radius: 0.25rem;
  color: #721c24;
  min-height: 0;
  padding: 0.75rem 1.25rem;
}

.notification-close {
  color: #000;
  font-size: 1.5rem;
  font-weight: 700;
  line-height: 1;
  opacity: 0.5;
  text-shadow: 0 1px 0 #fff;
}

.notification-progressbar {
  background-color: #7e6b6d;
  bottom: 0;
  height: 3px;
}
