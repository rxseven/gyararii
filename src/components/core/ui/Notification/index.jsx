import React from 'react';
import { ToastContainer } from 'react-toastify';

import './Notification.styles.scss';

function Close({ closeToast }) {
  return (
    <span
      aria-hidden="true"
      className="notification-close"
      onClick={closeToast}
    >
      &times;
    </span>
  );
}

function Notification(props) {
  return (
    <ToastContainer
      {...props}
      className="notification-container"
      closeButton={<Close />}
      progressClassName="notification-progressbar"
      toastClassName="notification-toast"
    />
  );
}

export default Notification;
