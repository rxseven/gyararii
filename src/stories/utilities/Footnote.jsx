import styled from 'styled-components';

const Footnote = styled.div`
  bottom: 2rem;
  font-size: 14px;
  position: fixed;
`;

export default Footnote;
