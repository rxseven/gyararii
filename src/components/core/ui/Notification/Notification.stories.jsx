import React, { useEffect } from 'react';
import { toast } from 'react-toastify';
import { storiesOf } from '@storybook/react';

import Notification from './index';

const title = 'Components/Core/UI/Notification';

function Toast() {
  useEffect(() => {
    toast.error('Notification message', {
      autoClose: false,
      position: toast.POSITION.TOP_LEFT
    });
  });

  return <Notification />;
}

storiesOf(title, module).add('default', () => <Toast />);
