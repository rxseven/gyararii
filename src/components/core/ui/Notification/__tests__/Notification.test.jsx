import React from 'react';
import { ToastContainer } from 'react-toastify';

import { factory } from 'tests/utilities';
import Notification from '../index';

// Mock
jest.mock('react-toastify', () => {
  return {
    ToastContainer: jest.fn(() => <i />)
  };
});

// Arrange
const source = {};

// Setup
function setup(props) {
  return factory(Notification, source, props);
}

// Test suites
describe('<Notification />', () => {
  it('should render without crashing', () => {
    setup();
  });

  it('should render a notification correctly', () => {
    setup();

    expect(ToastContainer).toHaveBeenCalled();
  });
});
