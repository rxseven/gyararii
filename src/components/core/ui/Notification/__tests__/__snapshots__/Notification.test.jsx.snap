// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> Snapshot tests should render correctly 1`] = `
<div
  className="Toastify"
/>
`;
